lang/elaborator/src/unifier/unify.rs

```diff
@@ -237,6 +237,22 @@ impl Ctx {
                     lhs: Box::new(lhs.clone()),
                     rhs: exp.clone(),
                 }),
+                (
+                    Exp::LocalComatch(LocalComatch { name: name_lhs, cases: cases_lhs, .. }),
+                    Exp::LocalComatch(LocalComatch { name: name_rhs, cases: cases_rhs, .. }),
+                ) if name_lhs == name_rhs => {
+                    let new_eqns =
+                        cases_lhs.iter().cloned().zip(cases_rhs.iter().cloned()).filter_map(
```

Comment:
I think this implementation depends on the order of the clauses being the same for both comatches. Even if we enforce this currently in the typechecker, it might no longer be true in the future.